From 3d35f806e7e3737dac5068323b55b042dbf2dde0 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Wed, 3 Jul 2024 19:54:33 +0500
Subject: [PATCH 109/157] net: ethernet: renesas: rswitch: fix crash if not all
 ports are enabled

rswitch_fwd_init() was trying to configure chains for all ports, without
checking if a port is enabled or not. That was causing a crash since
there is no rdev object for non-enabled port.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/drivers/net/ethernet/renesas/rswitch.c b/drivers/net/ethernet/renesas/rswitch.c
index bcb2f464a3dd..09ecb7d501bf 100644
--- a/drivers/net/ethernet/renesas/rswitch.c
+++ b/drivers/net/ethernet/renesas/rswitch.c
@@ -3276,6 +3276,8 @@ static void rswitch_fwd_init(struct rswitch_private *priv)
 
 	/* For ETHA ports, set port based forwarding to per-port rx chains */
 	for (i = 0; i < num_etha_ports; i++) {
+		if (!priv->etha[i].enabled)
+			continue;
 		/* Port-based forwarding to GWCA port */
 		rs_write32(FIELD_PREP(FWPBFC_PBDV_MASK, BIT(priv->gwca.index)),
 				priv->addr + FWPBFC(i));
-- 
2.25.1

